### PR TITLE
[Core] Adding constraints to FastTransferBetweenModelPartsProcess

### PR DESCRIPTION
**Pull request by loumalouomega**
Require #2889

### CODE REVIEW

File: kratos/processes/fast_transfer_between_model_parts_process.cpp [hunk 1]
```diff
@@ -151,6 +158,16 @@ void FastTransferBetweenModelPartsProcess::TransferWithFlags()
             }
         }
 
+        if (num_constraints != 0 && (mEntity == EntityTransfered::ALL || mEntity == EntityTransfered::CONSTRAINTS || mEntity == EntityTransfered::NODESANDCONSTRAINTS)) {
+            #pragma omp for
+            for(int i = 0; i < num_constraints; ++i) {
+                auto it_const = mrOriginModelPart.MasterSlaveConstraintsBegin() + i;
+                if (it_const->Is(mFlag)) {
+                    (constraints_buffer[id]).insert((constraints_buffer[id]).begin(), *(it_const.base()));
```
Comment (adityaghantasala) [inline kratos/processes/fast_transfer_between_model_parts_process.cpp]: a small thing for readability : id -> thread_id

**Comment by loumalouomega** [comment]
In any case this requires first #2889 to be approved

**Comment by adityaghantasala** [comment]
Another minor thing instead of `constraints_buffer` it can be `constraints_buffer_vector` representing what it actually is.

Anyway this looks consistent with the implementation for conditions and elements. 

One question independent of this PR : in the function ReplicateWithoutFlags you are reordering the ids.That means the original nodes/elements/conditions will not have the ids that they have from mdpa file or? wont this cause inconsistency with the input (mdpa) and the output(gid's .bin ) files ?

**Comment by adityaghantasala** [comment]
you can merge #2889

**Comment by loumalouomega** [comment]
Yes reordering will break the consistency with mdpa, but I don't know how to solve the problem without it

**Comment by RiccardoRossi** [comment]
what do you mean by "it will break the consistency with the mdpa"?

File: kratos/processes/fast_transfer_between_model_parts_process.cpp [hunk 2]
```diff
@@ -108,25 +113,27 @@ void FastTransferBetweenModelPartsProcess::TransferWithFlags()
...
     // Auxiliar sizes
     const int num_nodes = static_cast<int>(mrOriginModelPart.Nodes().size());
     const int num_elements = static_cast<int>(mrOriginModelPart.Elements().size());
     const int num_conditions = static_cast<int>(mrOriginModelPart.Conditions().size());
+    const int num_constraints = static_cast<int>(mrOriginModelPart.Conditions().size());
 
     #pragma omp parallel
     {
-        const int id = OpenMPUtils::ThisThread();
+        const int thread_id = OpenMPUtils::ThisThread();
```
Comment (RiccardoRossi) [inline kratos/processes/fast_transfer_between_model_parts_process.cpp:129]: do you really need the thread_id ? the best would be not to use this, unless it is strictly needed

Comment (loumalouomega) [reply kratos/processes/fast_transfer_between_model_parts_process.cpp:129]: Yes, in order to create a buffer for each thread. This is already done, just adding constraints

**Comment by loumalouomega** [comment]
Means that the ids of the nodes and elements will not be the same as the inputs ids of the mdpa

**Comment by RiccardoRossi** [comment]
...i would say that by default we should preserve the same ids...

**Comment by adityaghantasala** [comment]
If i remember the nodes / elements/ conditions are put in PointerVectorSet which is sorted. So it should be possible to find the max id and continue from there or ? This way it should be possible to preserve original ids.

**Comment by loumalouomega** [comment]
The reorder is done just in one specific case, when ReplicateWithoutFlags, and is because we use clone, which requires and ID as argument, and the only way to know for sure that the new id's well be coherent is to reorder.

**Comment by loumalouomega** [comment]
Replicate != Transfer, it is a clone

**Comment by RiccardoRossi** [comment]
the point is why you need to renumber...

**Comment by loumalouomega** [comment]
To be sure that there is not any id repetition (the model part could contain not consequent ids)

**Comment by loumalouomega** [comment]
Ping

**Comment by adityaghantasala** [comment]
From the constraints side the changes look fine. 
Reordering though it is not related to this PR, for seems to be a issue to be discussed further. It is ok if it is dealt with in a different PR or discussion. 
@RiccardoRossi can we do this like this that we discuss about reordering in a different PR and let this PR to continue or do you think this should also be resolved here ?

**Comment by loumalouomega** [comment]
Ping @RiccardoRossi

**Comment by loumalouomega** [comment]
Ping

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**